nautobot/extras/utils.py

```diff
@@ -615,82 +622,41 @@ def refresh_job_model_from_job_class(job_model_class, job_class, job_queue_class
-            "kind": "Pod",
-            "metadata": {"name": pod_name},
-            "spec": {
-                "containers": [
-                    {
-                        "image": nautobot_image_name,
-                        "name": nautobot_container_name,
-                    }
-                ]
-            },
-        }
-        # import yaml
-        # with open('./development/pod.yaml', 'r') as f:
-        #     pod_manifest = yaml.safe_load(f)
-        logger.info(f"Creating pod {pod_name} in namespace {pod_namespace}")
-        resp = api_instance.create_namespaced_pod(body=pod_manifest, namespace=pod_namespace)
-        logger.info(f"Reading pod {pod_name} in namespace {pod_namespace}")
-        resp = api_instance.read_namespaced_pod(name=pod_name, namespace=pod_namespace)
-        logger.info("Done.")
-
-    # Calling exec interactively
-    logger.info(f"Waiting for container {nautobot_container_name} to be fully set up")
-    exec_command = ["/bin/sh"]
-    resp = stream(
-        api_instance.connect_get_namespaced_pod_exec,
-        pod_name,
-        pod_namespace,
-        command=exec_command,
-        stderr=True,
-        stdin=True,
-        stdout=True,
-        tty=False,
-        _preload_content=False,
-    )
-    resp.write_stdin("nautobot-server runjob --local -u admin nautobot.core.jobs.ExportObjectList\n")
-    sresult = resp.read_stdout()
-    serror = resp.read_stderr()
-    logger.info(f"Job Result is: {sresult}")
-    logger.info(f"Error is: {serror}")
-    resp.close()
+    pod_manifest = copy.deepcopy(settings.KUBERNETES_JOB_POD_MANIFEST)
+    pod_ssl_ca_cert = settings.KUBERNETES_JOB_POD_SSL_CA_CERT
+    pod_token = settings.KUBERNETES_JOB_POD_TOKEN
+
+    configuration = kubernetes.client.Configuration()
+    configuration.host = settings.KUBERNETES_JOB_POD_HOST
+    configuration.ssl_ca_cert = pod_ssl_ca_cert
+    with open(pod_token, "r") as token_file:
+        token = token_file.read().strip()
+    # configure API Key authorization: BearerToken
+    configuration.api_key_prefix["authorization"] = "Bearer"
+    configuration.api_key["authorization"] = token
+    with kubernetes.client.ApiClient(configuration) as api_client:
+        api_instance = kubernetes.client.BatchV1Api(api_client)
+
+    job_result.task_kwargs = job_kwargs
+    job_result.save()
+    pod_manifest["metadata"]["name"] = "nautobot-job-" + str(job_result.pk)
+    pod_manifest["spec"]["template"]["spec"]["containers"][0]["command"] = [
+        "nautobot-server",
+        "runjob_with_job_result",
+        f"{job_result.pk}",
+    ]
+    logger.info("Creating job pod %s in namespace %s", pod_name, pod_namespace)
+    api_instance.create_namespaced_job(body=pod_manifest, namespace=pod_namespace)
+    logger.info("Reading job pod %s in namespace %s", pod_name, pod_namespace)
+    api_instance.read_namespaced_job(name="nautobot-job-" + str(job_result.pk), namespace=pod_namespace)
+    logger.info("Done.")
```

Comment:
Maybe remove this message - the job isn't actually done at this point, it's just been successfully enqueued right?